src/main/java/com/productdock/library/inventory/domain/BookSubscriptions.java

```diff
@@ -0,0 +1,49 @@
+package com.productdock.library.inventory.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Slf4j
+public class BookSubscriptions {
+
+    private String bookId;
+
+    private List<String> subscriberUserIds;
+
+    public void subscribeUser(String userId) {
+        initSubscribers();
+        if (!isUserSubscribed(userId)) {
+            subscriberUserIds.add(userId);
+        }
+    }
+
+    public void unsubscribeUser(String userId) {
+        if (subscriberUserIds != null) {
+            subscriberUserIds.remove(userId);
+        }
+    }
+
+    public boolean isUserSubscribed(String userId) {
+        if (subscriberUserIds != null) {
```

Comment:
I believe this null check is unnecessary. We call this method in two places, first is in subscribeUser, where you already initiated subscribers in the line above, and the second place is in checkSubscription method and there it's handled by orElse(false)